### ISSUE TITLE
STM8EF continued development

### ISSUE THREAD

**Issue by TG9541** [ForthHub/discussion#40]
I'm a Forth newbie working on an eForth for low-end ($0.20/piece) STM8S Value Line devices. My starting point was Dr. C.H. Ting's STM8EF eForth demo (a STC eForth). My contribution is new features, bug-fixes, support for cheap target boards, and more compact code (core, and compiled code). 

It's no match for AmForth or Mecrisp, but I like the challenge of targeting a device with very limited resources.

The repository is [here](https://github.com/TG9541/stm8ef). There is also a [Wiki](https://github.com/TG9541/stm8ef/wiki) with some documentation, and a [project on hackaday.io](https://hackaday.io/project/16097-eforth-for-cheap-stm8s-value-line-gadgets).

Feedback and advice from Forthers is welcome!

**Comment by MitchBradley** [comment]
This is probably not what you want to hear, but, for what it's worth, my advice is to focus on applications first, using needs to drive implementation features. 

Focusing on the system implementation can be fun, but it's almost always pointless now that there are so many implementations.

**Comment by TG9541** [comment]
Hi Mitch, well, I guess that's exactly the kind of advice I understand :-) 

The currently implemented system features are rather basic (simple multitasking, interrupt support, flash programming). I also provided support for several cheap Chinese boards (I/O, howtos). Following your advice I'll try to create simple "turnkey" application kits.

**Comment by lowfatcomputing** [comment]
Congrats, your project is featured on [hackaday](https://hackaday.com/2017/02/13/hacking-on-the-weirdest-esp-module/)!

**Comment by quozl** [comment]
Nice.  I would have put a Forth on the ESP8266 as well as the STM8, but oh well.  :grin:

**Comment by TG9541** [comment]
@lowfatcomputing Andreas, thanks! Several of the Hackaday seniors have been very supportive, and I owe them a big "thank you"! 

@quozl: have you tried [Punyforth](https://github.com/zeroflag/punyforth? In the heyday of Forth ESP8266 would have been a very very powerful device. The lowly STM8S is more like a janitor of the module :-)

 @MitchBradley nailed it: for making a Forth known one has to provide easy to use applications, not just the software. Cool "projects" that can be consumed easily by casual techies, youngsters and folks with a daytime job alike, are a good way to draw some attention to Forth. The [programmable $1.54 DC/DC converter](https://hackaday.io/project/19647-low-cost-programmable-power-supply) is a good example: it draws attention to Forth much more effectively than STM8EF that does the main job.

**Comment by quozl** [comment]
@TG9541, no, I use CForth on ESP8266.  Somewhat higher code to documentation ratio, but I read code faster than documentation.  :grin:
